src/content/structured/components/text-field/code.mdx

```diff
@@ -730,3 +730,168 @@ return (
     helperText="Such as Arabica, Robusta or Liberica"
   />
 </ComponentPreview>
+
+### Full width text field
+
+export const snippetsFullWidth = [
+  {
+    technology: "Web component",
+    snippets: {
+      short: `<ic-text-field  
+  label="What is your favourite coffee?" 
+  required="true" 
+  placeholder="Please enter…" 
+  helper-text="Such as Arabica, Robusta or Liberica" 
+  full-width="true"
+>
+</ic-text-field>`,
+      long: `.parent-container {
+    padding: var(--ic-space-md);
+  }
+</style>
+<body>
+  <div class="parent-container">
+    {shortCode}
+  </div>`,
+    },
+  },
+  {
+    technology: "React",
+    snippets: {
+      short: `<IcTextField
+  label="What is your favourite coffee?" 
+  required 
+  placeholder="Please enter…" 
+  helperText="Such as Arabica, Robusta or Liberica"
+  fullWidth="true"
+>
+</IcTextField>`,
+      long: [
+        {
+          language: "Typescript",
+          snippet: `const useStyles = createUseStyles({
+  parentContainer: {
+    padding: "var(--ic-space-md)",
+  },
+});
+const classes = useStyles();
+return (
+  <div className={classes.parentContainer}>
+    {shortCode}
+  </div>
+)`,
+        },
+        {
+          language: "Javascript",
+          snippet: `const useStyles = createUseStyles({
+  parentContainer: {
+    padding: "var(--ic-space-md)",
+  },
+});
+const classes = useStyles();
+return (
+  <div className={classes.parentContainer}>
+    {shortCode}
+  </div>
+)`,
+        },
+      ],
+    },
+  },
+];
+
+<ComponentPreview snippets={snippetsFullWidth}>
+  <IcTextField
+    label="What is your favourite coffee?"
+    required
+    placeholder="Please enter…"
+    helperText="Such as Arabica, Robusta or Liberica"
+    fullWidth="true"
+  />
+</ComponentPreview>
+
+### Custom width text field
+
+export const snippetsCustomWidth = [
+  {
+    technology: "Web component",
+    snippets: {
+      short: `<ic-text-field  
+  label="What is your favourite coffee?" 
+  required="true" 
+  placeholder="Please enter…" 
+  helper-text="Such as Arabica, Robusta or Liberica" 
+>
+</ic-text-field>`,
+      long: `.parent-container {
+    padding: var(--ic-space-md);
+  }
+  ic-text-field {
+    --input-width: 25.25rem;
+  }
+</style>
+<body>
+  <div class="parent-container">
+    {shortCode}
+  </div>`,
+    },
+  },
+  {
+    technology: "React",
+    snippets: {
+      short: `<IcTextField
+  label="What is your favourite coffee?" 
+  required 
+  placeholder="Please enter…" 
+  helperText="Such as Arabica, Robusta or Liberica"
+>
+</IcTextField>`,
+      long: [
+        {
+          language: "Typescript",
+          snippet: `const useStyles = createUseStyles({
+  parentContainer: {
+    padding: "var(--ic-space-md)",
+  },
+  textField: {
```

Comment:
This class setup isn't being applied in stackblitz as the class is never linked to the component. Better to take the example from the code preview where it uses the style attribute